### PR TITLE
[TestLoop Refactoring] (4/n) Further refactorings to allow ClientActions to be run in a TestLoop test

### PR DESCRIPTION
**Pull request by robin-near**
A few changes:
 - Split SyncJobsActor into SyncJobsActions/SyncJobsActor, the same way it was done for the ClientActor.
 - Replace the `Box<dyn Fn(...)>` with `Sender<...>` for the SyncJobsActor message senders given to ClientActions and other functions
 - Replace the sync jobs `ArbiterHandle` with a `FutureSpawner`. Added a `ActixArbiterHandleFutureSpawner` that spawns the future onto the `ArbiterHandle` rather than the spawner's arbiter.
 - Replace `Addr<TelemetryActor>` with `Sender<TelemetryEvent>`.
 - Introduced some test utils for making event listeners for TestLoop-based tests that use ClientActions and SyncJobsActions.
 - A few minor improvements for the TestLoop API, and the log visualizer UI.
 - A WIP example on testing Client with TestLoop. It doesn't do anything useful yet, but it's an attempt to get started. The major TODO here is to make the network layer work.

### CODE REVIEW

**Comment by robin-near** [comment]
FYI The next one is drafted here https://github.com/robin-near/nearcore/pull/3 It's a gigantic refactoring but it's basically just changing all of Duration, Instant, Utc to standarize on near_async::time::*, allowing the Clock to be used everywhere (as opposed to StaticClock which is used all over the place such as doomslug timer).

File: core/chain-configs/src/client_config.rs [hunk 1]
```diff
@@ -499,7 +499,7 @@ impl ClientConfig {
             num_block_producer_seats,
             ttl_account_id_router: Duration::from_secs(60 * 60),
             block_fetch_horizon: 50,
-            catchup_step_period: Duration::from_millis(1),
+            catchup_step_period: Duration::from_millis(100),
```
Comment (pugachAG) [inline core/chain-configs/src/client_config.rs:502]: is this intended change?

**Review by pugachAG — APPROVED** [review]
LGTM, consistent with the previous `[TestLoop Refactoring]` PRs

File: core/chain-configs/src/client_config.rs (hunk 1 above)
Comment (robin-near) [reply core/chain-configs/src/client_config.rs:502]: Yes, the previous 1ms doesn't seem to be used anywhere, and it causes the testloop based test to fire that timer every 1ms which is way too often.

**Pull request merged**